ouroboros-consensus/src/ouroboros-consensus/Ouroboros/Consensus/Storage/ChainDB/Impl/ChainSel.hs

```diff
@@ -735,12 +745,20 @@ chainSelectionForBlock cdb@CDB{..} blockCache hdr punish = do
          -- ^ The fragment with the new block @b@ as its tip, with the same
          -- anchor as @curChain@.
       -> Maybe Word64
-    computeLoEMaxExtra loeLimit loeFrag newBlockFrag
-      | budgetAlreadyUsed > loeLimit = Nothing
-      | otherwise                    = Just $ loeLimit - budgetAlreadyUsed
+    computeLoEMaxExtra loeLimit loeFrag newBlockFrag =
+        -- Both fragments are on the same chain
+        if loeSuffixLength == 0 || rollback == 0 then
+          if rollback > loeLimit + loeSuffixLength
+            then Nothing
+            else Just $ loeLimit + loeSuffixLength - rollback
+        else
+          if rollback > loeLimit
+            then Nothing
+            else Just $ loeLimit - rollback
```

Comment:
Currently, we disable the LoE by setting `loeLimit = maxBound`, so we should either refactor that or make sure we don't add to `loeLimit` without prior checks.

Reply:
The type of loeLimit was made isomorphic to Maybe in be94a46.